File: app/src/main/java/com/mstc/mstcapp/SplashActivity.java
package com.mstc.mstcapp;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.os.Handler;

import androidx.appcompat.app.AppCompatActivity;

import com.mstc.mstcapp.util.Constants;

public class SplashActivity extends AppCompatActivity {
    private final Context context = this;
    private SharedPreferences sharedPreferences;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        sharedPreferences = context.getSharedPreferences(Constants.STC_SHARED_PREFERENCES, MODE_PRIVATE);
        Handler handler = new Handler();
        handler.postDelayed(() -> {
            if (sharedPreferences.getBoolean("isFirstLaunch", true))
                startActivity(new Intent(context, WelcomeActivity.class));
            else
                startActivity(new Intent(context, MainActivity.class));
            finish();
        }, 500);

    }
}

File: app/src/main/java/com/mstc/mstcapp/SplashActivity.kt
package com.mstc.mstcapp

import android.app.NotificationManager
import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import com.mstc.mstcapp.util.Constants.Companion.STC_SHARED_PREFERENCES
import kotlinx.coroutines.*

class SplashActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        lifecycleScope.launch {
            val notificationManager =
                applicationContext.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
            notificationManager.cancelAll()
            delay(500)
            val sharedPreferences: SharedPreferences =
                getSharedPreferences(STC_SHARED_PREFERENCES, MODE_PRIVATE)
            startActivity(
                Intent(
                    this@SplashActivity,
                    when {
                        sharedPreferences.getBoolean(
                            "isFirstLaunch",
                            true
                        ) -> WelcomeActivity::class.java
                        else -> MainActivity::class.java
                    }
                )
            )
            finish()
        }
    }
}